From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: tizis0 <hexgames84@gmail.com>
Date: Mon, 19 Jan 2026 22:50:45 +0600
Subject: [PATCH] fix version_fetcher, add batching.mode config option,
 redesign metrics & realtime bar, add colors in /pulse command, add
 pulse.admin permission


diff --git a/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 3e0a328f361934464bc990c82070519017971938..c84599e27c1e12e23831388654c7e3b51b453833 100644
--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -368,7 +368,9 @@ public class ServerGamePacketListenerImpl
             }
         }
         // Pulse start
-        if (this.pulseBuffer != null) this.pulseBuffer.flush();
+        if (this.pulseBuffer != null) {
+            this.pulseBuffer.tick(); // new smart method for batching.mode
+        }
         // Pulse end
     }
 
